lint/vale_library.bzl

```diff
@@ -0,0 +1,112 @@
+        "homepage": "https://github.com/errata-ai/write-good",
+        "integrity": "sha256-4OhhIyZve4I3joSmGDu1XZjJthUovLc00XRy68Is1BU=",
+        "version": "v0.4.0",
+        "url": "https://github.com/errata-ai/write-good/releases/download/{}/write-good.zip",
+    },
+    {
+        "name": "alex",
+        "description": "Catch insensitive, inconsiderate writing.",
+        "homepage": "https://github.com/errata-ai/alex",
+        "integrity": "sha256-wREfIkw88Fag5g3GzuZ1OwDCOolVx9+VI/P29AtnePw=",
+        "version": "v0.2.1",
+        "url": "https://github.com/errata-ai/alex/releases/download/{}/alex.zip",
+    },
+    {
+        "name": "Readability",
+        "description": "Vale-compatible implementations of many popular readability metrics.",
+        "homepage": "https://github.com/errata-ai/Readability",
+        "integrity": "sha256-++F4tLZIxNQbtyzuUzrd/Y9Rzw89HwrfoIgIdwtBgQY=",
+        "version": "v0.1.1",
+        "url": "https://github.com/errata-ai/readability/releases/download/{}/Readability.zip",
+    },
+    {
+        "name": "Hugo",
+        "description": "Adds support for Hugo shortcodes and other non-standard markup.",
+        "homepage": "https://github.com/errata-ai/Hugo",
+        "integrity": "sha256-1xUjVXmEbh+sAeIKxruxwrLdEUz7M8SL0EysXUv+900=",
+        "version": "v0.2.0",
+        "url": "https://github.com/errata-ai/Hugo/releases/download/{}/Hugo.zip",
+    },
+    {
+        "name": "RedHat",
+        "description": "A Vale-compatible implementation of the Red Hat Supplementary Style Guide.",
+        "homepage": "https://redhat-documentation.github.io/vale-at-red-hat/docs/user-guide/redhat-style-for-vale/",
+        "integrity": "sha256-sZvjP/ahqSoua/OEdVC/oALUlFDXddut4pmKr5PWanU=",
+        "version": "v431",
+        "url": "https://github.com/redhat-documentation/vale-at-red-hat/releases/download/{}/RedHat.zip",
+    },
+    {
+        "name": "AsciiDoc",
+        "description": "A Vale-compatible implementation of select AsciiDoc syntax rules.",
+        "homepage": "https://redhat-documentation.github.io/vale-at-red-hat/docs/main/user-guide/asciidoc-style-for-vale/",
+        "integrity": "sha256-QbgpcmzACPYzpfEoUQx/69Pu73mqbgQnmqh3YQjY6hI=",
+        "version": "v431",
+        "url": "https://github.com/redhat-documentation/vale-at-red-hat/releases/download/{}/AsciiDoc.zip",
+    },
+    {
+        "name": "OpenShiftAsciiDoc",
+        "description": "A Vale-compatible implementation of select AsciiDoc guidance from the OpenShift docs contributor guidelines.",
+        "homepage": "https://redhat-documentation.github.io/vale-at-red-hat/docs/main/user-guide/openshift-asciidoc-style-for-vale/",
+        "integrity": "sha256-F6LMXFKfHIaFhG26BazlN3W2m1DN628f0YTe+An1TSQ=",
+        "version": "v431",
+        "url": "https://github.com/redhat-documentation/vale-at-red-hat/releases/download/{}/OpenShiftAsciiDoc.zip",
+    },
+]
+
+VALE_STYLES = [s["name"] for s in VALE_STYLE_DATA]
+
+def fetch_styles():
```

Comment:
Note: this api could accept an optional list of which styles to fetch, but I don't think it matters since unreferenced ones won't be fetched anyway.